### ISSUE TITLE
Opening Achievement Screen also calls API of leaderboard, contribution fragment

### ISSUE THREAD

**Issue by neeldoshii** [commons-app/apps-android-commons#6083]
Click on more, then from bottomsheet open Achievements. For some reason from logs you can see even being at Achievement Screen the leaderboard api is being called?

```
2024-12-26 21:03:22.611 OkHttpJsonApiClient     fr.free.nrw.commons                  I  Url https://tools.wmflabs.org/commons-android-app/tool-commons-android-app/leaderboard.py?user=Devanonymous&duration=weekly&category=upload&limit=100&offset=0
2024-12-26 21:03:22.611 OkHttp                  fr.free.nrw.commons                  V  --> GET https://tools.wmflabs.org/commons-android-app/tool-commons-android-app/leaderboard.py?user=Devanonymous&duration=weekly&category=upload&limit=100&offset=0
2024-12-26 21:03:22.611 OkHttp                  fr.free.nrw.commons                  V  --> END GET
```

Unable to share full logs due to github text limit. 

Take it up while working at #6063

**Comment by parneet-guraya** [comment]
That is intended, because in the `Profile` screen we use `Viewpager` and it has page limit of `1` by default so it instantiate neighboring fragment and fragment  does trigger a network request as it is instantiated. In this case opening `Achievements` fragment opens `Leaderboard` one too as it is next to it. and also if the fragment was in center it would have retained 1 on either side. Hope that makes sense :-)

![Screenshot from 2024-12-27 16-06-22](https://github.com/user-attachments/assets/b6f8a645-ca89-4e21-96fe-2730bd19b44f)

**Comment by neeldoshii** [comment]
> That is intended, because in the `Profile` screen we use `Viewpager` and it has page limit of `1` by default so it instantiate neighboring fragment and fragment  does trigger a network request as it is instantiated. In this case opening `Achievements` fragment opens `Leaderboard` one too as it is next to it. and also if the fragment was in center it would have retained 1 on each (L&R) side. Hope that makes sense :-)


Yes you are correct. Apologies, my bad I didn't wrote the description clear. I think we should migrate from Viewpager to Viewpager2 since it uses RecyclerView.Adapter so it dynamically allocates and deallocates the left and right one's. Wdyt?

**Comment by parneet-guraya** [comment]
I don't think it would be any different besides we want to have those pre instantiated for the faster context switches. 

Also, since we already are picking up Jetpack compose, I think we should use that instead.
